labs/a8.md

```diff
@@ -0,0 +1,736 @@
+lock in the address bar indicating that your website is secured with HTTPS!
+    
+For checkoff, please show that your have successfully requested and received a 
+signed certificate and have configured nginx to serve files using the 
+certificate and key to secure the HTTPS session.
+
+### File Security
+
+Now that you have some experience with the primitives of encryption, let's 
+cover some practical topics in securing files on UNIX-based systems.
+
+#### File Security and Permissions
+
+The base layer in the UNIX security hierarchy is file permissions. Every file 
+and process is owned by a user (and group), and by default, only the user/group 
+that owns the file can edit it. You can see this by typing `ls -la` in your 
+terminal:
+
+
+    admin@staff:~$ ls -la
+    total 104
+    drwxr-xr-x 9 admin admin  4096 Oct  3 13:01 .
+    drwxr-xr-x 5 root  root   4096 Oct  2 16:49 ..
+    drwxr-xr-x 2 admin admin  4096 Sep 21 21:11 .augeas
+    -rw------- 1 admin admin 27932 Oct  6 14:05 .bash_history
+    -rw-r--r-- 1 admin admin   220 May 15 12:45 .bash_logout
+    -rw-r--r-- 1 admin admin  3526 May 15 12:45 .bashrc
+    drwx------ 3 admin admin  4096 Sep 17 12:02 .config
+    drwxr-xr-x 4 admin admin  4096 Oct  3 12:47 .gem
+    drwxr-xr-x 2 admin admin  4096 Oct  3 12:46 .nano
+    -rw-r--r-- 1 admin admin   675 May 15 12:45 .profile
+    drwxr-xr-x 4 admin admin  4096 Sep 17 14:23 .puppet
+    drwx------ 2 admin admin  4096 Sep 17 12:09 .ssh
+    drwxr-xr-x 3 admin admin  4096 Oct  3 12:38 test
+    -rw------- 1 admin admin 21980 Oct  2 16:42 .viminfo
+
+The first column, e.g. `-rw-------`, is the read/write/execute permissions for 
+the file. The second and third columns are the user and group that own the file.
+
+Let's break down the permissions seen in the example above. You'll notice that 
+there are 10 entries: they either start with `d` for directory or `-` for 
+regular files[^firstpermflag], the the remaining 9 entries are split into 3 
+groups of 3 permission each: `r`ead, `w`rite, and e`x`ecute for `u`user, 
+`g`roup, and `o`ther. That means `-rw-------` refers to a regular file, where 
+the owner can read and write but not execute the file, and where the owning 
+group and everyone else can neither read, write, nor execute the file.
+On the other hand, the permissions on the `test` entry (`drwxr-xr-x`) indicate
+that it is a directory, everyone can enter the directory (`d--x--x--x`), anyone
+can list files inside the directory (`dr--r--r--`), but only the owning user can
+write files inside the directory (`d-w-------`).
+
+The kernel enforces file permissions, preventing running programs from reading 
+or modifying files they aren't allowed to, and preventing users from executing 
+programs that they don't have access or permission for. This is important for a 
+variety of reasons. For example, certain UNIX user accounts have their 
+passwords stored in a file called `/etc/passwd` or `/etc/shadow`. These files 
```

Comment:
For practical purposes, passwords are hashed and stored in `/etc/shadow`; `/etc/passwd` is world-readable and contains nonsensitive account information.